### ISSUE TITLE
GHA jobs failing for `RESTHybridTopologyFactory`

### ISSUE THREAD

**Issue by zhang-ivy** [choderalab/perses#895]
The tests for `RESTCapableHybridTopologyFactory` (https://github.com/choderalab/perses/pull/848) are not passing for some of the GHA jobs: https://github.com/choderalab/perses/runs/4319599038?check_suite_focus=true

Two problems: 

1. The jobs that aren't passing are the ones that use OpenMM 7.5.1 and the `ubuntu-latest, Python 3.9, OpenMM nightly`job. 
Looking at the "Environment Information" for the aforementioned OpenMM nightly job, its actually using 7.5.1: 
```
openmm                    7.5.1            py39h71eca04_1    conda-forge
```
Not sure why this is happening.

2. The reason why the 7.5.1 jobs are failing is because `RESTCapableHybridTopologyFactory` uses `setName()` to set the name of `CustomNonbondedForce` and `CustomBondForce`, since there are more than one of them in the hybrid sytsem. However, [OpenMM 7.5.1](https://github.com/openmm/openmm/releases/tag/7.5.1) doesn't have `setName()` for the `CustomNonbondedForce` / `CustomBondForce`. That method is introduced in [7.6.0](https://github.com/openmm/openmm/releases/tag/7.6.0beta). Here is the error I am getting:
```
  def _create_electrostatics_force(self):
        """
        Create `CustomNonbondedForce` for direct space PME electrostatics.
        """
    
        import itertools
    
        # Create the force
        expression = self._default_electrostatics_expression
        formatted_expression = expression.format(alpha_ewald=self._alpha_ewald,
                                                 w_scale=self._w_scale,
                                                 r_cutoff=self._r_cutoff.value_in_unit_system(unit.md_unit_system))
        custom_force = openmm.CustomNonbondedForce(formatted_expression)
        name = custom_force.__class__.__name__ + '_electrostatics'
>       custom_force.setName(name)
E       AttributeError: 'CustomNonbondedForce' object has no attribute 'setName'

perses/annihilation/relative.py:3708: AttributeError
```
Not sure what to do about this -- should we stop testing 7.5.1?

**Comment by ijpulidos** [comment]
One of the issues is that we had to pin the OpenMM version to 7.5.1 when 7.6 was released as a workaround with the import problems. This was probably a bad quick fix.

Does this mean we now only support OpenMM 7.6 if users wants to use the RESTHybridTopologyFactory ?

**Comment by mikemhenry** [comment]
Yes there isn't python3.9 in the nightly dev builds, see this issue https://github.com/openmm/openmm/issues/3320

I think we need to decide what version(s) of openmm we support. We can set up the codebase to disable some features if the version of openmm is too old, but that increases maintaince. On the other hand, if we only support the newest version of openmm, that could mess up some users workflow who need an older version of openmm, we should talk about this today at our meeting so I will add it to the milestone so we can discus it.

**Comment by mikemhenry** [comment]
This will be fixed in #896
